src/tests/ftest/dfuse/daos_build.py

```diff
@@ -49,41 +50,87 @@ def test_daos_build(self):
 
-        daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
-                                    attr='dfuse-ndentry-time', val='60s')
+        cont_attrs = OrderedDict()
+
+        cache_mode = self.params.get('name', '/run/dfuse/*')
+
+        # How long to cache things for, if caching is enabled.
+        cache_time = '30m'
+        build_time = 15
+
+        if cache_mode == 'writeback':
+            cont_attrs['dfuse-data-cache'] = 'on'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+        elif cache_mode == 'writethrough':
+            cont_attrs['dfuse-data-cache'] = 'on'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+        elif cache_mode == 'metadata':
+            cont_attrs['dfuse-data-cache'] = 'off'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+        elif cache_mode == 'nocache':
+            build_time = 150
+            cont_attrs['dfuse-data-cache'] = 'off'
+            cont_attrs['dfuse-attr-time'] = '0'
+            cont_attrs['dfuse-dentry-time'] = '0'
+            cont_attrs['dfuse-ndentry-time'] = '0'
+
+        for key, value in cont_attrs.items():
+            daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
+                                        attr=key, val=value)
 
         self.start_dfuse(self.hostlist_clients, self.pool, self.container)
 
         mount_dir = self.dfuse.mount_dir.value
         build_dir = os.path.join(mount_dir, 'daos')
 
-        cmds = ['git clone https://github.com/daos-stack/daos.git {}'.format(build_dir),
-                'git -C {} submodule init'.format(build_dir),
-                'git -C {} submodule update'.format(build_dir),
-                '{} -C {} --jobs 50 build --build-deps=yes'.format(scons, build_dir)]
+        intercept = self.params.get('use_intercept', '/run/intercept/*', default=False)
+
+        remote_env = OrderedDict()
+        remote_env['PATH'] = f'{mount_dir}/venv/bin:$PATH'
+        remote_env['VIRTUAL_ENV'] = f'{mount_dir}/venv'
+
+        if intercept:
+            remote_env['LD_PRELOAD'] = '/usr/lib64/libioil.so'
+            remote_env['D_LOG_FILE'] = '/var/tmp/daos_testing/daos-il.log'
+            remote_env['DD_MASK'] = 'all'
+            remote_env['DD_SUBSYS'] = 'all'
+            remote_env['D_LOG_MASK'] = 'INFO,IL=DEBUG'
+
+        envs = []
+        for env, value in remote_env.items():
+            envs.append('export {}={}'.format(env, value))
+
+        preload_cmd = ';'.join(envs)
+
+        cmds = [f'python3 -m venv {mount_dir}/venv',
+                f'git clone https://github.com/daos-stack/daos.git {build_dir}',
```

Comment:
Possibly.  I guess this works in CI as it's in our base images but might fail if run outside?

Reply:
Ahh.  That's how I discovered https://github.com/daos-stack/daos/pull/8789#pullrequestreview-961287629.  I went to see how it could have worked without the `Requires: git`.

Reply:
That's a link back to this PR.